epam_tr/src/main/java/com/epam/tr/service/FileSystemService.java

```diff
@@ -0,0 +1,145 @@
+        checkFile(fileEntity);
+        try {
+            if (fileEntity.getType() == FOLDER) {
+                createFolder(fileEntity);
+            }
+            if (fileEntity.getType() == FILE) {
+                createFile(fileEntity);
+            }
+        } catch (IOException e) {
+            throw new WrongFileException(CREATE_EXCEPTION);
+        }
+    }
+
+    @Override
+    public void update(FileEntity oldFile, FileEntity newFile) throws WrongFileException {
+        checkFile(oldFile);
+        checkFile(newFile);
+        String oldFileAbsolutePath = createAbsolutePath(oldFile);
+        String newFileAbsolutePath = createAbsolutePath(newFile);
+        new File(oldFileAbsolutePath).renameTo(new File(newFileAbsolutePath));
+    }
+
+    @Override
+    public void delete(FileEntity fileEntity) throws WrongFileException {
+        checkFile(fileEntity);
+        String absolutePath = createAbsolutePath(fileEntity);
+        new File(absolutePath).delete();
+    }
+
+    private void checkFile(FileEntity fileEntity) throws WrongFileException {
+        if (!validator.isValid(fileEntity)) {
+            throw new WrongFileException(NOT_VALID);
+        }
+    }
+
+    private void createFile(FileEntity fileEntity) throws IOException {
+        String absolutePath = createAbsolutePath(fileEntity);
+        new File(absolutePath).createNewFile();
+    }
+
+    private void createFolder(FileEntity fileEntity) {
+        new File(fileEntity.getPath(), fileEntity.getName()).mkdir();
+    }
+
+    private String createAbsolutePath(FileEntity fileEntity) {
+        String path = fileEntity.getPath();
+        String name = fileEntity.getName();
+        String absolutePath = path + "\\" + name;
+        if (fileEntity.getType() == FILE) {
+            absolutePath += ".txt";
+        }
+        return absolutePath;
+    }
+
+    @Override
+    public List<FileEntity> readFileByPath(String path) {
+        List<FileEntity> files;
+        if (path.isEmpty()) {
+            files = getRoots();
+        } else {
+            File dir = new File(path);
+            files = scanDir(dir);
+        }
+        return files;
+    }
+
+    @Override
+    public List<FileEntity> filter(String path, String parameter, String sortType) {
+        List<FileEntity> list = readFileByPath(path);
+        Comparator<FileEntity> sorter = SorterFactory.createSorter(parameter);
+        if (sortType.equals(ASC)) {
+            list.sort(sorter);
+        } else if (sortType.equals(DESC)) {
+            list.sort(sorter.reversed());
+        }
+        return list;
+    }
+
+    @Override
+    public List<FileEntity> search(String path, String mask) {
+        List<FileEntity> fileList = this.readFileByPath(path);
+        Pattern pattern = Pattern.compile(mask);
+        List<FileEntity> suitableFiles = new ArrayList<>();
+        fileList.stream().filter((fileEntity) -> pattern.matcher(fileEntity.getName()).matches()).forEach((suitableFiles::add));
```

Comment:
hhmmm. that is strange, did you forget about collect method?
use it instead of forEach